qmi/instruments/newport/smc_100cc.py

```diff
@@ -30,3 +44,542 @@ def __init__(self,
+        if persist:
+            self.reset(controller_address)
+            self.enter_configuration_state(controller_address)
+
+        self._scpi_protocol.write(self._build_command(
+            "FD", frequency))
+        sleep(self.COMMAND_EXEC_TIME)
+        self._check_error()
+        if persist:
+            self.exit_configuration_state(controller_address)
+
+    @rpc_method
+    def get_following_error_limit(self, controller_address: Optional[int] = None) -> float:
+        """
+        Get the maximum allowed following error.
+
+        Parameters:
+            controller_address: Optional address of the controller that needs to be controlled. By default,
+                                it is set to the initialised value of the controller address.
+
+        Returns:
+            The value for the maximum allowed following error.
+        """
+        _logger.info(
+            "Getting encoder increment value of instrument [%s]", self._name)
+        # instrument must be in configuration state to get the encoder increment value.
+        self.reset(controller_address)
+        self.enter_configuration_state(controller_address)
+        res = self._scpi_protocol.ask(
+            self._build_command("FE?"))
+        sleep(self.COMMAND_EXEC_TIME)
+        self._check_error()
+        self.exit_configuration_state(controller_address)
+        return float(res[3:])
+
+    @rpc_method
+    def set_following_error_limit(
+            self, error_limit: float, persist: bool = False, controller_address: Optional[int] = None) -> None:
+        """
+        Set the value for the maximum allowed following error.
+
+        Parameters:
+            error_limit:        The value for the maximum allowed following error.
+            persist:            Flag to indicate if the error limit should be persisted to the controller's memory, so
+                                it is still available after powering down the controller. When not persisted, the error
+                                limit is the one stored in the controller's memory.
+            controller_address: Optional address of the controller that needs to be controlled. By default,
+                                it is set to the initialised error limit of the controller address.
+        """
+        if 1E12 <= error_limit or error_limit <= 1E-6:
+            raise QMI_InstrumentException(
+                f"Provided value {error_limit} not in valid range 1E-6 > error limit > 1E12.")
+
+        _logger.info(
+            "Setting the value for the maximum allowed following error of instrument [%s] to [%f]",
+            self._name, error_limit
+        )
+        self.controller_address = controller_address
+        # instrument must be in configuration state to persist the set velocity.
+        if persist:
+            self.reset(controller_address)
+            self.enter_configuration_state(controller_address)
+
+        self._scpi_protocol.write(self._build_command(
```

Comment:
Same ass the `FD` command. It cannot be called in the Ready state. So you would have to make sure the controller is in the Disable state here

Reply:
I made now checks to all relevant functions like this to check first the state, and if it is READY, set it to DISABLE with the `MM` command. If it is already DISABLE, go ahead. For `get` functions we enter CONFIGURATION state if the current state was neither READY nor DISABLE. For `set` functions the same if no `persist` KW is present, but if it is and it is `True` then we go to the CONFIGURATION state directly. If `persist` is `False` then we do check the READY/DISABLE, but raise an exception if the state is not correct to set the value without. 

A little extra note is that similar approach is also applied to other commands which accept bot the READY and DISABLE state, next to CONFIGURATION to `get/set`. Only that no `MM` command will be called.